crates/hyperqueue/src/client/commands/journal/report.rs

```diff
@@ -0,0 +1,832 @@
+use crate::common::format::human_duration;
+use crate::common::utils::time::parse_human_time;
+use crate::server::autoalloc::AllocationId;
+use crate::server::event::journal::JournalReader;
+use crate::server::event::payload::EventPayload;
+use crate::transfer::messages::{JobTaskDescription, SubmitRequest};
+use anyhow::anyhow;
+use chrono::{DateTime, Duration, TimeDelta, Utc};
+use clap::{Parser, ValueHint};
+use itertools::Itertools;
+use std::collections::HashMap;
+use std::ops::Add;
+use std::path::PathBuf;
+use tako::gateway::{ResourceRequest, ResourceRequestVariants};
+use tako::resources::ResourceAmount;
+use tako::worker::WorkerConfiguration;
+use tako::{JobId, JobTaskId, ResourceVariantId, TaskId, WorkerId};
+
+#[derive(Parser)]
+pub(crate) struct JournalReportOpts {
+    /// Path to a journal
+    #[arg(value_hint = ValueHint::FilePath)]
+    journal: PathBuf,
+
+    /// Path to an output file
+    #[arg(value_hint = ValueHint::FilePath)]
+    output: PathBuf,
+
+    /// Show only statistics from a given time
+    ///
+    /// Counted from the start of the server (e.g. "1h")
+    #[arg(long)]
+    start_time: Option<String>,
+
+    /// Show only statistics upto a given time
+    ///
+    /// Counted from the start of the server (e.g. "1h")
+    #[arg(long)]
+    end_time: Option<String>,
```

Comment:
We can use this:
```rust
#[arg(
        long,
        value_parser = parse_hms_or_human_time,
        help = duration_doc!("Duration after which will an idle worker automatically stop")
    )]
    pub idle_timeout: Option<Duration>,
```
for parsing human durations.

Reply:
Fixed